source/fundamentals/data-formats/pojo-customization.txt

```diff
@@ -10,78 +10,653 @@ POJO Customization
+
+For more information on this class, see the :java-docs:`ClassModel </apidocs/bson/org/bson/codecs/pojo/ClassModel.html>`
+API documentation.
+
+To instantiate a ``ClassModel`` use the ``ClassModel.builder()`` method and
+specify your POJO class. The builder uses reflection to create the required
+metadata.
+
+.. code-block:: java
+   :copyable: true
+
+   ClassModel<Flower> classModel = ClassModel.builder(Flower.class).build();
+
+.. _property-model:
 
 PropertyModel
--------------
+~~~~~~~~~~~~~
+
+A ``PropertyModel`` stores information on how to serialize/deserialize a
+specific field in a document.
+
+The ``PropertyModel`` contains the following information:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Field Name
+     - Description
+
+   * - Name
+     - Specifies the property name in the model.
+
+   * - Read Name
+     - Name of the property to use as the key when serializing to BSON.
+
+   * - Write Name
+     - Name of the property to use as the key when deserializing from BSON.
+
+   * - Type data
+     - Contains an instance of ``org.bson.codecs.pojo.TypeData`` that
+       describes the data type for the field.
+
+   * - Codec
+     - Specifies a codec to use to encode or decode the field. Optional.
+
+   * - Serialization checker
+     - Determines whether to serialize a value by using the criteria
+       specified in the checker.
 
-- Description
-- Table of properties
-- Link to Conventions section for example of PropertyModelBuilder
+   * - Property accessor
+     - Method used to access the value of the property from the POJO
+
+   * - useDiscriminator
+     - Specifies whether to use the discriminator.
+
+To create a ``PropertyModel`` use a ``PropertyModelBuilder`` which you
+can instantiate by calling the ``PropertyModel.builder()`` method.
+
+For more information, see the
+:java-docs:`PropertyModel.Builder API documentation <apidocs/bson/org/bson/codecs/pojo/PropertyModelBuilder.html>`.
+
+
+.. _conventions:
 
 Conventions
 -----------
 
-- Description
-- Table of preset Conventions
-  - Link to Annotations section
-- Example on how to specify a Convention
-- How to create a custom Convention
+The ``Convention`` interface contains configuration options that modify the
+behavior of a ``ClassModel`` or ``PropertyModel``. You can specify a
+``Convention`` in a call to ``PojoCodecProvider.Builder.conventions()`` or to
+``ClassModelBuilder.conventions()``. The builders apply ``Convention``
+instances in order which may override behavior defined in one applied earlier.
+
+You can access the ``Convention`` instances defined in the BSON library
+from the following static fields in the ``Conventions`` class:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Field Name
+     - Description
+
+   * - ``ANNOTATION_CONVENTION``
+     - Enables the annotations defined in the ``org.bson.codecs.pojo.annotations``
+       package for your POJO.
```

Comment:
I know that the Driver docs tend to leave links to the end of a section, but I wonder if it would be useful to link to the list of those annotations here (since this description isn't hugely helpful without knowing the annotations you can use!). It seems like a link to the next section would be adequate? Up to you.

Reply:
Thanks, I'll link to the next section. To reiterate why API documentation goes to the end of the section is because we received both writer and reader feedback that opening links external to the doc distract from the material.